packages/tiny-router/src/actions.ts

```diff
@@ -138,17 +190,49 @@ export const beforeSSR: TinyRouter["actions"]["router"]["beforeSSR"] = ({
   state,
   actions,
 }) => async ({ ctx }) => {
-  if (state.router.autoFetch) {
-    if (actions.source && actions.source.fetch) {
-      await actions.source.fetch(state.router.link);
-      const data = state.source.get(state.router.link);
-      if (isError(data)) {
-        ctx.status = data.errorStatus;
-      }
-    } else {
-      warn(
-        "You are trying to use autoFetch but no source package is installed."
-      );
+  // If autoFetch is disabled, there is nothing to do.
+  if (!state.router.autoFetch) {
+    return;
+  }
+
+  // Because Frontity is a modular framework, it could happen that a source
+  // package like `@frontity/wp-source` has not been installed but the user is
+  // trying to use autoFetch option, which requires it.
+  if (!actions.source || !actions.source.fetch || !state.source.get) {
+    warn("You are trying to use autoFetch but no source package is installed.");
+    return;
+  }
+
+  await actions.source.fetch(state.router.link);
+  const data = state.source.get(state.router.link);
+
+  // Check if the link has a redirection.
+  if (data && isRedirection(data)) {
+    // If the redirection is external, just redirect to the full URL here.
+    if (data.isExternal) {
+      ctx.redirect(data.location);
+      return;
+    }
+
+    // Recover all the missing query params from the original URL. This is
+    // required because we remove the query params that start with `frontity_`.
+    const location = new URL(data.location, "https://dummy-domain.com");
```

Comment:
Same here. Maybe I'm wrong... :thinking:

Reply:
Explained in https://github.com/frontity/frontity/pull/601#discussion_r563676453.